iothub_client/inc/iothub_client_core_common.h

```diff
@@ -150,14 +150,80 @@ extern "C"
 
     MU_DEFINE_ENUM_WITHOUT_INVALID(DEVICE_TWIN_UPDATE_STATE, DEVICE_TWIN_UPDATE_STATE_VALUES);
 
+/**
+ * @brief Current versions of @p IOTHUB_CLIENT_GET_TWIN_REQUEST_OPTIONS structure.
+ */
+#define IOTHUB_CLIENT_GET_TWIN_REQUEST_OPTIONS_STRUCT_VERSION_1 1
+
+/**
+ * @brief   GET twin request options
+ */
+typedef struct IOTHUB_CLIENT_GET_TWIN_REQUEST_OPTIONS_TAG
+{
+    /**
+     * @brief   Version of the structure.
+     *
+     * @note    Currently must be @p IOTHUB_CLIENT_GET_TWIN_REQUEST_OPTIONS_STRUCT_VERSION_1.
+     * @note    Memory for this structure will be allocated by the application and used by the SDK,
+     *          which may be built as a dynamic library. Because the structure may grow, to avoid
+     *          the SDK inadvertently accessing non-struct-allocated memory, the struct version set
+     *          by the application will be checked by the SDK.
+     */
+    unsigned int struct_version;
+
+    /**
+     * @brief   Is the @p current_twin_version option to be used.
+     *
+     * @note    True if `current_twin_version` is to be used. False if `current_twin_version` is not
+     *          to be used.
+     * @note    If true, `current_twin_version` must also be set by the application.
+     */
+    bool is_current_twin_version_set;
+
+    /**
+     * @brief   The device client's current version for the twin properties section being requested.
+     *
+     * @note    To use this option, @p is_current_twin_version_set must be true.
+     * @note    If the IoT Hub's version of the requested section matches, no payload will be
+     *          returned and the response status will be 304. If the IoT Hub's version does not
+     *          match, the IoT Hub will return the requested twin section. If this option is not
+     *          used at all (`is_current_twin_version_set` == false), the IoT Hub will return the
+     *          requested twin section.
+     * @note    Valid range is [1, 9223372036854775807].
+     */
+    int64_t current_twin_version;
+
+} IOTHUB_CLIENT_GET_TWIN_REQUEST_OPTIONS;
+
+/**
+ * @brief   GET twin response
+ *
+ * @note    Memory for this structure will be allocated by the SDK and used by the application.
+ *          Therefore no struct versioning is required.
+ */
+typedef struct IOTHUB_CLIENT_GET_TWIN_RESPONSE_TAG
+{
+    /**
+     * @brief   The status set from the response topic.
+     */
+    int64_t status;
+
+    /**
+     * @brief   The $version set from the response topic.
+     */
+    int64_t version;
```

Comment:
twin_version ?

Reply:
Did we talk about making these 'current_device_version' and 'version'?  SInce each will be within a struct labeled twin_tequest _properties or twin_response, the twin part would be implied...?